sound/soc/sof/sof-pci-dev.c

```diff
@@ -205,42 +205,11 @@ static int sof_pci_probe(struct pci_dev *pci,
 #else
 	/* find machine */
 	mach = snd_soc_acpi_find_machine(desc->machines);
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_HDA)
 	if (!mach) {
-		dev_warn(dev, "No matching ASoC machine driver found - falling back to HDA codec\n");
-		mach = snd_soc_acpi_intel_hda_machines;
-		mach->sof_fw_filename = desc->nocodec_fw_filename;
-
-		/*
-		 * TODO: we need to find a way to check if codecs are actually
-		 * present
-		 */
+		dev_warn(dev, "warning: No matching ASoC machine driver found\n");
 	}
-#endif /* CONFIG_SND_SOC_SOF_HDA */
-
-#if IS_ENABLED(CONFIG_SND_SOC_SOF_NOCODEC)
-	if (!mach) {
-		/* fallback to nocodec mode */
-		dev_warn(dev, "No matching ASoC machine driver found - using nocodec\n");
-		mach = devm_kzalloc(dev, sizeof(*mach), GFP_KERNEL);
-		if (!mach) {
-			ret = -ENOMEM;
-			goto release_regions;
-		}
-		ret = sof_nocodec_setup(dev, sof_pdata, mach, desc, ops);
-		if (ret < 0)
-			goto release_regions;
-	}
-#endif /* CONFIG_SND_SOC_SOF_NOCODEC */
-
 #endif /* CONFIG_SND_SOC_SOF_FORCE_NOCODEC_MODE */
 
-	if (!mach) {
-		dev_err(dev, "error: no matching ASoC machine driver found - aborting probe\n");
-		ret = -ENODEV;
-		goto release_regions;
```

Comment:
you need to check the error flow. now there is no longer an error when no machine driver is found, so this is a new flow which needs to be checked

Reply:
yes, I will refine it

Reply:
I check the error flow and find that there is only one error label release_regions which is shared by many error checks. I will check the status in sof-probe and release_regions there.